docs/site/Binding.md

```diff
@@ -83,6 +83,36 @@ binding.toDynamicValue(() => new Date());
 binding.toDynamicValue(() => Promise.resolve('my-value'));
 ```
 
+The factory function can receive extra information about the context, binding,
+and resolution options.
+
+```ts
+import {ValueFactory} from '@loopback/context';
+
+// The factory function now have access extra metadata about the resolution
+const factory: ValueFactory<string> = resolutionCtx => {
```

Comment:
Is there any particular reason against using object destructuring here? I think it's a pretty common pattern that most JS/TS developers should be familiar with.

```suggestion
const factory: ValueFactory<string> = ({context, binding, options}) => {
```

Reply:
IMO, it's better to show the original parameter type for documentation purpose.

Reply:
I added another example with object destructuring.